PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/RemoteDebugRunConfiguration.java

```diff
@@ -22,17 +22,107 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
 
 public class RemoteDebugRunConfiguration extends RunConfigurationBase {
+    private static final String SUBMISSION_CONTENT_NAME = "spark_submission";
+    private static final String SUBMISSION_ATTRIBUTE_CLUSTER_NAME = "cluster_name";
+    private static final String SUBMISSION_ATTRIBUTE_IS_LOCAL_ARTIFACT = "is_local_artifact";
+    private static final String SUBMISSION_ATTRIBUTE_ARTIFACT_NAME = "artifact_name";
+    private static final String SUBMISSION_ATTRIBUTE_FILE_PATH = "file_path";
+    private static final String SUBMISSION_ATTRIBUTE_CLASSNAME = "classname";
+
+    private SparkSubmitModel submitModel;
+
     public RemoteDebugRunConfiguration(@NotNull Project project, @NotNull ConfigurationFactory factory, String name) {
         super(project, factory, name);
+
+        this.submitModel = new SparkSubmitModel(project);
+    }
+
+    @Override
+    public void readExternal(Element rootElement) throws InvalidDataException {
+        super.readExternal(rootElement);
+
+        SparkSubmitModel model = getSubmitModel();
+        SparkSubmissionParameter parameter = Optional.ofNullable(model.getSubmissionParameter())
+                .orElse(new SparkSubmissionParameter(
+                        "",
+                        false,
+                        "",
+                        "",
+                        "",
+                        "",
+                        new ArrayList<String>(),
+                        new ArrayList<String>(),
+                        new ArrayList<String>(),
+                        Arrays.stream(SparkSubmissionParameter.defaultParameters)
+                            .collect(Collectors.toMap(Pair::first, Pair::second))));
+
+        Optional.ofNullable(rootElement.getChild(SUBMISSION_CONTENT_NAME)).ifPresent((element -> {
+            Optional.ofNullable(element.getAttribute(SUBMISSION_ATTRIBUTE_CLUSTER_NAME))
+                    .ifPresent(attribute -> parameter.setClusterName(attribute.getValue()));
+            Optional.ofNullable(element.getAttribute(SUBMISSION_ATTRIBUTE_IS_LOCAL_ARTIFACT))
+                    .ifPresent(attribute -> parameter.setLocalArtifact(attribute.getValue().toLowerCase()                                                              .equals("true")));
+            Optional.ofNullable(element.getAttribute(SUBMISSION_ATTRIBUTE_ARTIFACT_NAME))
+                    .ifPresent(attribute -> parameter.setArtifactName(attribute.getValue()));
+            Optional.ofNullable(element.getAttribute(SUBMISSION_ATTRIBUTE_CLASSNAME))
+                    .ifPresent(attribute -> parameter.setClassName(attribute.getValue()));
+
+            model.setSubmissionParameters(parameter);
+        }));
+    }
+
+    @Override
+    public void writeExternal(Element rootElement) throws WriteExternalException {
```

Comment:
Not for here. Here is Run configuration, what should refer to http://www.jetbrains.org/intellij/sdk/docs/basics/run_configurations/run_configuration_management.html#persistence